core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/signer/DefaultAwsV4HttpSigner.java

```diff
@@ -157,7 +182,7 @@ private static V4PayloadSigner v4PayloadSigner(
             throw new UnsupportedOperationException("Unsigned payload is not supported with event-streaming.");
         }
 
-        if (isChunkEncoding) {
+        if ((isChunkEncoding && isPayloadSigning) || (isChunkEncoding && (isTrailing || isFlexible))) {
```

Comment:
Suggesting extracting it to a separate method and add javadoc to make it more clear
```

private boolean useChunkedEncoding(BaseSignRequest<?, ? extends AwsCredentialsIdentity> request) 
```